### ISSUE TITLE
jetbrains-toolbox: fails to start, appimage-exec.sh -w jetbrains-toolbox-extracted not such file or directory

### ISSUE THREAD

**Issue by starquake** [NixOS/nixpkgs#233446]
### Describe the bug
jetbrains-toolbox: fails to start
```console
[nix-shell:~/]$ jetbrains-toolbox

/nix/store/4s2bb3m6904nxd7l188bva1hdnpnq1qi-jetbrains-toolbox-1.28.1.15219-init: line 19: /home/starquake/Development/nixos-configuration/appimage-exec.sh -w /nix/store/qx9098ximvzqks5nr4z5i6raa24jvl7z-jetbrains-toolbox-extracted --: No such file or directory
```

### Steps To Reproduce
1. `nix-shell -p jetbrains-toolbox`
2. run `jetbrains-toolbox`

### Expected behavior
Jetbrains Toolbox window should show up

### Additional context
Recent install of NixOS. Upgraded to unstable. Trying to install jetbrains toolbox I get this error.

If there is anything else I can do, please let me know!

### Notify maintainers
@AnatolyPopov 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
starquake@animal:~$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.29, NixOS, 23.05 (Stoat), 23.05pre486734.50a183182d7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(starquake): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

**Comment by dacioromero** [comment]
I don't think this is relegated to just jetbrains-toolbox. I've had identical issues with other AppImage apps like protonup-qt which I found when trying to wrap an AppImage myself.

**Comment by jnsgruk** [comment]
Yep - I'm also seeing this for `rambox`. I wonder if something changed in the underlying appimage package tools.

**Comment by jnsgruk** [comment]
Related to https://github.com/NixOS/nixpkgs/issues/233327

**Comment by jnsgruk** [comment]
Looking around in the `appimage` package, not much seems to have changed in the `appimage-exec.sh` for over 6 months, but last month there was a tree-wide move from `buildFHSUserEnv` -> `buildFHSEnv` and I wonder if that was responsible? 

The commit against the `appimage` package is [here](https://github.com/NixOS/nixpkgs/commit/f63a12f296b806a1b838d6fd8eef99fa65929649). 

That said, I can't seem to correlate the timing of that change and when `rambox` stopped working for me, so I wonder if there has been a change to `appImageTools` used when building the package?

**Comment by gaykitty** [comment]
I'm having the same problem with the program sonixd, which is also an appimage. I tracked the breakage down to this change, in the file call `<hash>-sonixd-0.15.5-init`. The final line changes from

```
exec appimage-exec.sh -w /nix/store/v0m3rzah3z2qghr9dls05gwn8pzjczq5-sonixd-0.15.5-extracted -- "$@"
```
to
```
exec 'appimage-exec.sh -w /nix/store/v0m3rzah3z2qghr9dls05gwn8pzjczq5-sonixd-0.15.5-extracted --' "$@"
```
If the single quotes around the call to `appimage-exec.sh` are remove, the program is able to launch again. I have not yet figured out what change in the nix source lead to this change in the outputted script.

**Comment by gaykitty** [comment]
This was fixed in master with [#233209](https://github.com/NixOS/nixpkgs/pull/233209)
